src/chains/ethereum/ethereum/src/forking/fork.ts

```diff
@@ -227,11 +234,13 @@ export class Fork {
       this.blockNumber.toBigInt()
     );
     this.block = new Block(BlockManager.rawFromJSON(block, common), common);
-    if (!chainOptions.time && minerOptions.timestampIncrement !== "clock") {
-      chainOptions.time = new Date(
-        (this.block.header.timestamp.toNumber() +
-          minerOptions.timestampIncrement.toNumber()) *
-          1000
+    if (!chainOptions.time) {
+      // this needs to set the time of the _previous_ block,
+      // whereas in incrementBasedBlockTime we set the time of the _next_ block
+      // maybe when we call setTime, we increment the value passed in.
+      const blockTimestampMs = this.block.header.timestamp.toNumber() * 1000;
+      this.block.header.timestamp = Quantity.from(
```

Comment:
Ah. Good to know.

I've set that value, and validated (note the timezone difference!):
```
Forked Chain
==================
Location:        Ethereum Mainnet, via 丕Infura
Block:           15073145
Network ID:      1
Time:            Mon Jul 04 2022 13:41:06 GMT+1200 (New Zealand Standard Time)

Chain Id
==================
1337
```

Reply:
Except the timestamp is now incorrect if ganache is using "clock" mode (since it doesn't use the clock for the first block after the fork block).

Note: I don't particularly like that this is ganache's behavior in "clock" mode.

Reply:
I've added some explicit tests around this behaviour to `forking.test.ts`